app/uk/gov/hmrc/economiccrimelevyregistration/navigation/Navigator.scala

```diff
@@ -42,6 +43,14 @@ class Navigator @Inject() () {
       case _           => routes.StartController.onPageLoad()
     }
 
+  private def amlSupervisorRoute(registration: Registration): Call =
+    registration.amlSupervisor match {
+      case Some(AmlSupervisor(GamblingCommission, _)) | Some(AmlSupervisor(FinancialConductAuthority, _)) =>
+        routes.RegisterWithOtherAmlSupervisorController.onPageLoad()
+      case Some(AmlSupervisor(Hmrc, _)) | Some(AmlSupervisor(Other, _))                                   => routes.EntityTypeController.onPageLoad()
```

Comment:
Given we don't care about the `otherProfessionalBody` - we can just match on the `AmlSupervisorType` to reduce some noise here, i.e. we don't need to match on the whole case class

Reply:
I thought that originally as well but couldn't figure out how to match on just the AmlSupervisorType :)

Reply:
More specifically, because this is an Option I couldn't easily get to the supervisor type without forcibly getting the AmlSupervisor (which I thought was "unsafe", i.e. we can handle the fact that it might be None, or an unexpected supervisor type, in the match case if we matched on the entire case class)

Reply:
Sure, if you wanted to match on just the `AmlSupervisorType` instead of the whole `Option[AmlSupervisor]` you could do it like this:

```
    registration.amlSupervisor match {
      case Some(amlSupervisor) => amlSupervisor.supervisorType match {
        case FinancialConductAuthority | GamblingCommission => routes.RegisterWithOtherAmlSupervisorController.onPageLoad()
        case Hmrc | Other => routes.EntityTypeController.onPageLoad()
      }
      case _                                                                                              => routes.StartController.onPageLoad()
    }
```

I suppose it just depends what you think reads better - I think this probably reads a little better

Reply:
Updated match logic as suggested.